catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java

```diff
@@ -92,39 +123,193 @@ public void initialize(Map<String, String> config, CatalogInfo info) throws Runt
   }
 
   @Override
   public Topic createTopic(
       NameIdentifier ident, String comment, DataLayout dataLayout, Map<String, String> properties)
       throws NoSuchSchemaException, TopicAlreadyExistsException {
-    throw new UnsupportedOperationException();
+    NameIdentifier schemaIdent = NameIdentifier.of(ident.namespace().levels());
+    if (!schemaExists(schemaIdent)) {
+      LOG.warn("Kafka catalog schema {} does not exist", schemaIdent);
+      throw new NoSuchSchemaException("Schema %s does not exist", schemaIdent);
+    }
+
+    try {
+      CreateTopicsResult createTopicsResult =
+          adminClient.createTopics(Collections.singleton(buildNewTopic(ident, properties)));
+      LOG.info(
+          "Created topic {} with {} partitions and replication factor {}",
+          ident,
+          createTopicsResult.numPartitions(ident.name()).get(),
+          createTopicsResult.replicationFactor(ident.name()).get());
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof TopicExistsException) {
+        throw new TopicAlreadyExistsException(e, "Topic %s already exists", ident);
+
+      } else if (e.getCause() instanceof InvalidReplicationFactorException) {
+        throw new IllegalArgumentException(
+            "Invalid replication factor for topic " + ident + e.getCause().getMessage(), e);
+
+      } else if (e.getCause() instanceof InvalidConfigurationException) {
+        throw new IllegalArgumentException(
+            "Invalid properties for topic " + ident + e.getCause().getMessage(), e);
+
+      } else {
+        throw new RuntimeException("Failed to create topic in Kafka" + ident, e);
+      }
+    } catch (InterruptedException e) {
+      throw new RuntimeException("Failed to create topic in Kafka" + ident, e);
+    }
+
+    return KafkaTopic.builder()
+        .withName(ident.name())
+        .withComment(comment)
+        .withProperties(properties)
+        .withAuditInfo(
+            AuditInfo.builder()
+                .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                .withCreateTime(Instant.now())
+                .build())
+        .build();
   }
 
   @Override
   public Topic alterTopic(NameIdentifier ident, TopicChange... changes)
       throws NoSuchTopicException, IllegalArgumentException {
-    throw new UnsupportedOperationException();
+    KafkaTopic topic = (KafkaTopic) loadTopic(ident);
+    String newComment = topic.comment();
+    int oldPartitionCount = Integer.parseInt(topic.properties().get(PARTITION_COUNT));
+    int newPartitionCount = oldPartitionCount;
+    Map<String, String> alteredProperties = Maps.newHashMap(topic.properties());
+    List<AlterConfigOp> alterConfigOps = Lists.newArrayList();
+    for (TopicChange change : changes) {
+      if (change instanceof TopicChange.UpdateTopicComment) {
+        newComment = ((TopicChange.UpdateTopicComment) change).getNewComment();
+
+      } else if (change instanceof TopicChange.SetProperty) {
```

Comment:
So Kafka does not support changing the number of `replications`?

Reply:
Please see https://github.com/datastrato/gravitino/pull/2615#discussion_r1540428128